### PR TITLE
Implement NEO3 VM stack opcodes

### LINKED ISSUE #876: Implement Neo 3 VM stack opcodes
Implement new stack manipulation opcodes, neo-project/neo-vm#237, neo-project/neo-vm#258, both in VM and compiler.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #876 .

### CODE REVIEW

File: pkg/vm/vm.go [hunk 1]
```diff
@@ -736,6 +736,15 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 			panic(err.Error())
 		}
 
+	case opcode.REVERSE3, opcode.REVERSE4, opcode.REVERSEN:
+		n := 3 + int(op-opcode.REVERSE3)
```
Comment (roman-khimov) [inline pkg/vm/vm.go:719]: I'd rather not obfuscate `REVERSE4` here this way.

File: pkg/vm/opcode/opcode.go [hunk 2]
```diff
@@ -65,31 +65,36 @@ const (
...
+	NIP   Opcode = 0x46
+	XDROP Opcode = 0x48
+	// CLEAR    Opcode = 0x49
+	DUP      Opcode = 0x4A
+	OVER     Opcode = 0x4B
+	PICK     Opcode = 0x4D
+	TUCK     Opcode = 0x4E
+	SWAP     Opcode = 0x50
+	OLDPUSH1 Opcode = 0x51 // FIXME remove #xxx
+	ROT      Opcode = 0x7B
```
Comment (roman-khimov) [inline pkg/vm/opcode/opcode.go:80]: I think we can make `OLDPUSH` an alias of new `ROT` which is `0x51` and then make a kludge in `Context` to treat one-byte script containing `ROT` as `PUSH1`. This way we won't have to update scripts again after `OLDPUSH1` removal.

File: pkg/vm/vm.go (hunk 1 above)
Comment (fyrchik) [reply pkg/vm/vm.go]: Fixed.

File: pkg/vm/opcode/opcode.go (hunk 2 above)
Comment (fyrchik) [reply pkg/vm/opcode/opcode.go]: Yeah, it is actually sufficient.

**Pull request merged**